pkg/sdn/apis/network/validation/validation.go

```diff
@@ -25,36 +25,43 @@ func SetDefaultClusterNetwork(cn sdnapi.ClusterNetwork) {
+		allErrs = append(allErrs, field.Invalid(field.NewPath("clusterNetworks"), clusterNet.ClusterNetworks, "must have at least one cluster network CIDR"))
 	}
+	for _, cn := range clusterNet.ClusterNetworks {
+		clusterIPNet, err := netutils.ParseCIDRMask(cn.CIDR)
+		if err != nil {
+			allErrs = append(allErrs, field.Invalid(field.NewPath("network"), cn.CIDR, err.Error()))
+		} else {
+			maskLen, addrLen := clusterIPNet.Mask.Size()
+			if cn.HostSubnetLength > uint32(addrLen-maskLen) {
+				allErrs = append(allErrs, field.Invalid(field.NewPath("hostSubnetLength"), cn.HostSubnetLength, "subnet length is too large for clusterNetwork"))
+			} else if cn.HostSubnetLength < 2 {
+				allErrs = append(allErrs, field.Invalid(field.NewPath("hostSubnetLength"), cn.HostSubnetLength, "subnet length must be at least 2"))
+			}
+		}
 
-	if (clusterIPNet != nil) && (serviceIPNet != nil) && clusterIPNet.Contains(serviceIPNet.IP) {
-		allErrs = append(allErrs, field.Invalid(field.NewPath("serviceNetwork"), clusterNet.ServiceNetwork, "service network overlaps with cluster network"))
-	}
-	if (serviceIPNet != nil) && (clusterIPNet != nil) && serviceIPNet.Contains(clusterIPNet.IP) {
-		allErrs = append(allErrs, field.Invalid(field.NewPath("network"), clusterNet.Network, "cluster network overlaps with service network"))
+		serviceIPNet, err := netutils.ParseCIDRMask(clusterNet.ServiceNetwork)
+		if err != nil {
+			allErrs = append(allErrs, field.Invalid(field.NewPath("serviceNetwork"), clusterNet.ServiceNetwork, err.Error()))
+		}
+		if (clusterIPNet != nil) && (serviceIPNet != nil) && clusterIPNet.Contains(serviceIPNet.IP) {
+			allErrs = append(allErrs, field.Invalid(field.NewPath("serviceNetwork"), clusterNet.ServiceNetwork, "service network overlaps with cluster network"))
+		}
+		if (serviceIPNet != nil) && (clusterIPNet != nil) && serviceIPNet.Contains(clusterIPNet.IP) {
+			allErrs = append(allErrs, field.Invalid(field.NewPath("network"), clusterNet.Network, "cluster network overlaps with service network"))
+		}
 	}
 
 	if clusterNet.Name == sdnapi.ClusterNetworkDefault && defaultClusterNetwork != nil {
-		if clusterNet.Network != defaultClusterNetwork.Network {
-			allErrs = append(allErrs, field.Invalid(field.NewPath("network"), clusterNet.Network, "cannot change the default ClusterNetwork record via API."))
-		}
-		if clusterNet.HostSubnetLength != defaultClusterNetwork.HostSubnetLength {
-			allErrs = append(allErrs, field.Invalid(field.NewPath("hostSubnetLength"), clusterNet.HostSubnetLength, "cannot change the default ClusterNetwork record via API."))
+		if len(clusterNet.ClusterNetworks) == len(defaultClusterNetwork.ClusterNetworks) {
+			for index := range defaultClusterNetwork.ClusterNetworks {
+				if clusterNet.ClusterNetworks[index].CIDR != defaultClusterNetwork.ClusterNetworks[index].CIDR || clusterNet.ClusterNetworks[index].HostSubnetLength != defaultClusterNetwork.ClusterNetworks[index].HostSubnetLength {
```

Comment:
We do not want to bail out when the order of cluster networks changed. 
Lets say, OldClusterNetwork=[cidrA, cidrB] and newClusterNetwork=[cidrB, cidrA]. Here cidrA and cidrB order has changed but the config itself did not change, so there should not be any error in this case.